tools/saqc/saqc.xml

```diff
@@ -344,50 +319,73 @@
+              <param argument="flag" type="float" value="255.0" optional="true" label="flag" help="`Any`, optional The flag value the function uses to mark observations. Defaults to the ``BAD`` value of the translation scheme"/>
             </when>
             <when value="clearFlags">
-              <param argument="field" type="text" value="" label="Field" help="The name of the variable to process.">
-                <validator type="empty_field"/>
-              </param>
+              <param argument="field" type="data_column" label="field" help="Variable to process" data_ref="data" multiple="false"/>
             </when>
             <when value="flagDummy">
-              <param argument="field" type="text" value="" label="Field" help="The name of the variable to process.">
-                <validator type="empty_field"/>
-              </param>
+              <param argument="field" type="data_column" label="field" help="Variable to process" data_ref="data" multiple="false"/>
             </when>
             <when value="flagManual">
-              <param argument="field" type="text" value="" label="Field" help="The name of the variable to process.">
-                <validator type="empty_field"/>
-              </param>
-              <param argument="mdata" type="text" value="" label="Determines which values or intervals will be flagged" help="Supported input types:  * ``pd.Series``: Needs a datetime index and values of type:  - datetime, for `method` values `` right-closed ``, `` left-closed ``, `` closed `` - or any scalar, for `method` values `` plain ``, `` ontime ``  * ``str``: Variable holding the manual flag information. * ``pd.DataFrame``, ``DictOfSeries``: Need to provide a ``pd.Series`` with column name `field`. * ``list``, ``np.ndarray``: Only supported with `method` value `` plain `` and `mformat` value `` mflag ``">
-                <validator type="empty_field"/>
-              </param>
+              <param argument="field" type="data_column" label="field" help="Variable to process" data_ref="data" multiple="false"/>
+              <conditional name="mdata_cond">
+                <param name="mdata_selector" type="select" label="Choose type for 'mdata'" help="Determines which values or intervals will be flagged. Supported input types:  * ``pd.Series``: Needs a datetime index and values of type:  - datetime, for `method` values `` right-closed ``, `` left-closed ``, `` closed `` - or any scalar, for `method` values `` plain ``, `` ontime ``  * ``str``: Variable holding the manual flag information. * ``pd.DataFrame``, ``DictOfSeries``: Need to provide a ``pd.Series`` with column name `field`. * ``list``, ``np.ndarray``: Only supported with `method` value `` plain `` and `mformat` value `` mflag ``">
```

Comment:
This is an interesting case: The Supported input types are separately listed in the help. Do you think it would be reasonable to split it into the 4 ways to input data. 

We have a list, where the 1st word is equal to the type annotation. This could be parsed?